application/src/main/java/org/opentripplanner/transit/model/timetable/ScheduledTripTimes.java

```diff
@@ -146,7 +146,7 @@ public int getServiceCode() {
   }
 
   @Override
-  public ScheduledTripTimes withServiceCode(int serviceCode) {
+  public TripTimes withServiceCode(int serviceCode) {
```

Comment:
Why don't you guarantee a return of `ScheduledTripTimes` here? This will avoid the need to change `FrequencyEntry` as well.